File: back/Main.Core/Entities/Equipment.cs
using Main.Core.Entities.Geometry;
using Shared.Core.Entities;

namespace Main.Core.Entities;

public class Equipment : Entity<Equipment>
{
    public required string Name { get; set; }
    public required string Status { get; set; } = "no-data";

    public EquipmentPoint? Feature { get; set; }
    public Guid? FeatureId { get; set; }

    public EquipmentType? EquipmentType { get; set; }
    public required Guid EquipmentTypeId { get; set; }

    public Organization? Organization { get; set; }
    public required Guid OrganizationId { get; set; }

    public Department? Department { get; set; }
    public Guid? DepartmentId { get; set; }
    
    public override void Update(Equipment entity)
    {
        Name = entity.Name;
        Status = entity.Status;

        FeatureId = entity.FeatureId;
        EquipmentTypeId = entity.EquipmentTypeId;
        OrganizationId = entity.OrganizationId;
        DepartmentId = entity.DepartmentId;
    }
}

File: front/src/shared/misc/types/entities/equipment/Equipment.ts
import { Feature, Point } from "@turf/turf";

import { Status } from "./Status";
import { Entity } from "../base";
import { Department } from "../department";
import { EquipmentType } from "../equipment-type";
import { Organization } from "../organization";

export interface Equipment extends Entity {
  name: string;
  status: Status;
  feature: Feature<Point>;
  type: EquipmentType;
  organization: Organization;
  department?: Department;
}
